### ISSUE TITLE
cue/load: package cycles are not detected

### ISSUE THREAD

**Issue by cueckoo** [cue-lang/cue#849]
_Originally opened by @grouville in https://github.com/cuelang/cue/issues/849_

### What version of CUE are you using (`cue version`)?

<pre>
$ cue version
cue version 0.3.0-beta.7 darwin/amd64 (installed with brew)
</pre>

### Does this issue reproduce with the latest release?
Yes

### What did you do?

It seems that `cue eval` / `cue export` goes into an infinite loop when evaluating/exporting packages with cyclic inclusion

Here is a tree representation of a little implementation, to reproduce the error :
```bash
➜  test_infinite tree
.
├── cue.mod
│   ├── module.cue
│   ├── pkg
│   │   └── top.dev
│   │       ├── la
│   │       │   └── toto.cue
│   │       └── lo
│   │           └── tata.cue
│   └── usr
└── main.cue
```
---------
*main.cue* 
```cue
package main

import (
	"top.dev/lo"
)

test: {
    value: #JO & {
        test2: "top"
    }
}
```
---------
*module.cue* 
```cue
module: "main"
```
---------
*toto.cue*
```cue
package la

import (
	"top.dev/lo"
)

#LA: {
	test?: string
}
```
---------
*tata.cue*
```cue
package lo

import (
	"top.dev/la"
)

#JO: {
	test2?: string
}
```
----------
----------

*output*
```bash
➜  test time cue export
^C
cue export  325,33s user 8,33s system 173%!c(MISSING)pu 3:12,37 total
```

### What did you expect to see?

I'm not sure this is an issue. I think it is, but I might be wrong :
- With a broken field in the main (other test, same structure. Change the main.cue like that :
```cue
package main

import (
	"top.dev/lo"
)

test: {
    value: #JO & {
        test2: 1
    }
}
```
=> With this main, a validation error (test2 expects a string, and I provide an int)

- With the first main showed, a warning, or maybe not an infinite inclusion [I'm not sure if this is an expected behavior]

### What did you see instead?

An infinite loop

**Comment by cueckoo** [comment]
_Original reply by @myitcv in https://github.com/cuelang/cue/issues/849#issuecomment-804755067_

@grouville - thanks for raising your first issue in the CUE project!

Just as an FYI, there are some tips on how to create clean reproducers in https://github.com/cuelang/cue/wiki/Creating-test-or-performance-reproducers. The `txtar` format is particularly good when it comes to multi-file repros, as is the case here. 

For example, here is my reproducer based on the above:

```
exec cue eval

-- cue.mod/module.cue --
module: "main.com"
-- cue.mod/pkg/top.dev/la/toto.cue --
package la

import (
	"top.dev/lo"
)

#LA: {
	test?: string
}
-- cue.mod/pkg/top.dev/lo/tata.cue --
package lo

import (
	"top.dev/la"
)

#JO: {
	test2?: string
}
-- main.cue --
package main

import (
	"top.dev/lo"
)

test: {
	value: #JO & {
		test2: "top"
	}
}
```

As you can see, a `cue eval` in the module root is sufficient to reproduce this issue.

The issue is that we are not detecting package cycles during the load phase: there is a cycle `top.dev/la <-> top.dev/lo`. Package cycles are illegal, but should obviously be better detected and reported. 

Thanks for raising this - I'm slightly surprised we don't have a test case that covers this. Either way, we'll investigate and fix.

**Comment by cueckoo** [comment]
_Original reply by @grouville in https://github.com/cuelang/cue/issues/849#issuecomment-804915884_

@myitcv Thank you for the tip regarding the clean reproducers. I didn't know about this format, and it is far cleaner :-)

**Comment by cueckoo** [comment]
_Original reply by @myitcv in https://github.com/cuelang/cue/issues/849#issuecomment-805773959_

Note this also failed in `v0.2.2` - so we will fix this after the release of `v0.3.0` (so as not to delay that any longer)

**Comment by myitcv** [comment]
The v0.5.x milestone is now closed because we have moved onto focus on v0.6.0. Hence moving this to v0.6.x. This is being looked at in https://github.com/cue-lang/cue/pull/2435.

**Comment by bozaro** [comment]
Is there a chance to get a fix in 0.6.0?
